client/trino-client/src/main/java/io/trino/client/uri/TrinoUri.java

```diff
@@ -0,0 +1,1249 @@
+        this.kerberosServicePrincipalPattern = KERBEROS_SERVICE_PRINCIPAL_PATTERN.getValueOrDefault(urlProperties, kerberosServicePrincipalPattern);
+        this.kerberosRemoteServiceName = KERBEROS_REMOTE_SERVICE_NAME.getValueOrDefault(urlProperties, kerberosRemoteServiceName);
+        this.kerberosUseCanonicalHostname = KERBEROS_USE_CANONICAL_HOSTNAME.getValueOrDefault(urlProperties, kerberosUseCanonicalHostname);
+        this.kerberosPrincipal = KERBEROS_PRINCIPAL.getValueOrDefault(urlProperties, kerberosPrincipal);
+        this.kerberosConfigPath = KERBEROS_CONFIG_PATH.getValueOrDefault(urlProperties, kerberosConfigPath);
+        this.kerberosKeytabPath = KERBEROS_KEYTAB_PATH.getValueOrDefault(urlProperties, kerberosKeytabPath);
+        this.kerberosCredentialCachePath = KERBEROS_CREDENTIAL_CACHE_PATH.getValueOrDefault(urlProperties, kerberosCredentialCachePath);
+        this.kerberosDelegation = KERBEROS_DELEGATION.getValueOrDefault(urlProperties, kerberosDelegation);
+        this.kerberosConstrainedDelegation = KERBEROS_CONSTRAINED_DELEGATION.getValueOrDefault(urlProperties, kerberosConstrainedDelegation);
+        this.accessToken = ACCESS_TOKEN.getValueOrDefault(urlProperties, accessToken);
+        this.externalAuthentication = EXTERNAL_AUTHENTICATION.getValueOrDefault(urlProperties, externalAuthentication);
+        this.externalAuthenticationTimeout = EXTERNAL_AUTHENTICATION_TIMEOUT.getValueOrDefault(urlProperties, externalAuthenticationTimeout);
+        this.externalRedirectStrategies = EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS.getValueOrDefault(urlProperties, externalRedirectStrategies);
+        this.externalAuthenticationTokenCache = EXTERNAL_AUTHENTICATION_TOKEN_CACHE.getValueOrDefault(urlProperties, externalAuthenticationTokenCache);
+        this.extraCredentials = EXTRA_CREDENTIALS.getValueOrDefault(urlProperties, extraCredentials);
+        this.hostnameInCertificate = HOSTNAME_IN_CERTIFICATE.getValueOrDefault(urlProperties, hostnameInCertificate);
+        this.clientInfo = CLIENT_INFO.getValueOrDefault(urlProperties, clientInfo);
+        this.clientTags = CLIENT_TAGS.getValueOrDefault(urlProperties, clientTags);
+        this.traceToken = TRACE_TOKEN.getValueOrDefault(urlProperties, traceToken);
+        this.sessionProperties = SESSION_PROPERTIES.getValueOrDefault(urlProperties, sessionProperties);
+        this.source = SOURCE.getValueOrDefault(urlProperties, source);
+
+        properties = buildProperties();
+
+        // enable SSL by default for the trino schema and the standard port
+        useSecureConnection = SSL.getValue(properties).orElse(uri.getScheme().equals("https") || (uri.getScheme().equals("trino") && uri.getPort() == 443));
+        if (!password.orElse("").isEmpty()) {
+            if (!useSecureConnection) {
+                throw new SQLException("Authentication using username/password requires SSL to be enabled");
```

Comment:
```suggestion
                throw new SQLException("TLS/SSL required for authentication with username and password");
```

Reply:
But can that not be disabled with the insecure config?

Reply:
The `--insecure` flag in the CLI is the equivalent of `SSLVerification=NONE`, so it doesn't disable TLS/SSL. In the CLI, enabling TLS is done by specifying the `https` protocol or using the `443` port if there's no protocol.

I changed the error message to `TLS/SSL is required for authentication with username and password`, with the extra `is` before `required`.

Reply:
Insecure means "accept certificate that is not trusted"

Reply:
I applied the suggested error.